satpy/readers/modis_l1b.py

```diff
@@ -62,130 +84,158 @@ class HDFEOSBandReader(HDFEOSBaseFileReader):
-                                'toa_bidirectional_reflectance')
-            elif key['calibration'] == 'radiance':
-                projectable = calibrate_radiance(array, var_attrs, index)
-                info.setdefault('units', var_attrs.get('radiance_units'))
-                info.setdefault('standard_name',
-                                'toa_outgoing_radiance_per_unit_wavelength')
-            elif key['calibration'] == 'counts':
-                projectable = calibrate_counts(array, var_attrs, index)
-                info.setdefault('units', 'counts')
-                info.setdefault('standard_name', 'counts')  # made up
-            else:
-                raise ValueError("Unknown calibration for "
-                                 "key: {}".format(key))
-            projectable.attrs = info
-
-            # if ((platform_name == 'Aqua' and key['name'] in ["6", "27", "36"]) or
-            #         (platform_name == 'Terra' and key['name'] in ["29"])):
-            #     height, width = projectable.shape
-            #     row_indices = projectable.mask.sum(1) == width
-            #     if row_indices.sum() != height:
-            #         projectable.mask[row_indices, :] = True
-
-            # Get the orbit number
-            # if not satscene.orbit:
-            #     mda = self.data.attributes()["CoreMetadata.0"]
-            #     orbit_idx = mda.index("ORBITNUMBER")
-            #     satscene.orbit = mda[orbit_idx + 111:orbit_idx + 116]
-
-            # Trimming out dead sensor lines (detectors) on terra:
-            # (in addition channel 27, 30, 34, 35, and 36 are nosiy)
-            # if satscene.satname == "terra":
-            #     for band in ["29"]:
-            #         if not satscene[band].is_loaded() or satscene[band].data.mask.all():
-            #             continue
-            #         width = satscene[band].data.shape[1]
-            #         height = satscene[band].data.shape[0]
-            #         indices = satscene[band].data.mask.sum(1) < width
-            #         if indices.sum() == height:
-            #             continue
-            #         satscene[band] = satscene[band].data[indices, :]
-            #         satscene[band].area = geometry.SwathDefinition(
-            #             lons=satscene[band].area.lons[indices, :],
-            #             lats=satscene[band].area.lats[indices, :])
-            self._add_satpy_metadata(key, projectable)
-            return projectable
+            return subdata, uncertainty, var_attrs, band_index
+
+    def _get_band_index(self, var_attrs, band_name):
+        """Get the relative indices of the desired channel."""
+        band_names = var_attrs["band_names"].split(",")
+        index = band_names.index(band_name)
+        return index
+
+    def _mask_invalid_or_fill_saturated(self, array, valid_range):
+        """Replace fill values with NaN or replace saturation with max reflectance."""
+        # Fill values:
```

Comment:
Could we put this in the docstring instead?

Reply:
Yes. I'll see if I can get it to format nicely.